drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/AutomaticPseudoClock.java

```diff
@@ -42,10 +42,24 @@ public void shutdown() {
     }
 
     protected void advancePseudoClock() {
-        nextTick += period;
-        if (Math.abs(System.currentTimeMillis() - nextTick) > period * 2) {
-            LOG.warn("Pseudo clock is diverged, the difference is {} ms", (System.currentTimeMillis() - nextTick));
+
+        long diff = System.currentTimeMillis() - nextTick;
+        if (diff > period * 2) {
+            LOG.warn("Pseudo clock is diverged, the difference is {} ms. Going to sync with the real clock.", diff);
+            // DROOLS-7569 : Do not rely on "catch up" mechanism of ScheduledThreadPoolExecutor.scheduleAtFixedRate().
+            // This is explicit "catch up" logic.
+            // Also do not leap to the current time at once, because it may miss some rule firings.
+            for (; nextTick < System.currentTimeMillis(); nextTick += period) {
+                rulesEvaluator.scheduledAdvanceTimeToMills(nextTick);
+            }
```

Comment:
Previous logic relied on "catch up" mechanism of `ScheduledThreadPoolExecutor.scheduleAtFixedRate()` which quickly adds a next task to the Executor when the pseudo clock is diverged. However, https://issues.redhat.com/browse/AAP-16664 tells that the mechanism doesn't work in a special occasion. Actually, such "catch up" mechanism is not stated in ScheduledThreadPoolExecutor javadoc.

So this PR does explicit "catch up" when the pseudo clock is diverged.